include/core/mir_toolkit/common.h

```diff
@@ -481,6 +481,21 @@ typedef enum MirOutputGammaSupported
     mir_output_gamma_supported
 } MirOutputGammaSupported;
 
+/**
+ * Depth layer controls Z ordering of surfaces.
+ * A surface will always appear on top of surfaces with a lower depth layer, and below those with a higher one.
+ */
+typedef enum MirDepthLayer
+{
+    mir_depth_layer_background,         /**< For desktop backgrounds and alike (lowest layer) */
+    mir_depth_layer_below,              /**< For panels or other controls/decorations below normal windows */
+    mir_depth_layer_application,        /**< For normal application windows */
+    mir_depth_layer_always_on_top,      /**< For always-on-top application windows */
```

Comment:
I'm unconvinced that this is best modeled as a layer in this scheme. For example: it's entirely possible that one would want an always-on-top window on `mir_depth_layer_background` - for example, you want your Conky desktop widget showing your system stats *on top* of any other background, but below everything else.

Reply:
That is exactly what `mir_depth_layer_below` is for. If you add an "always on top" boolean (as you seem to be advocating for), I don't see how that's different than just doubling the number of layers.

Reply:
I think that, on any given layer, it's perfectly reasonable for a surface to want to be on top of all other surfaces on that layer. This can't be expressed as doubling the number of layers, for fear of infinities :grin: